packages/docs/celo-codebase/protocol/identity/phone-number-privacy.md

```diff
@@ -0,0 +1,38 @@
+
+Quota is influenced by:
+- Account balance
+- Transaction history
+- Phone number attestation count and success rate
+
+### Matchmaking
+To hinder potential data harvesting, the lookup quotas should be kept small. However, users may want to discover which of there large set of contacts are on the Celo network. To enable that scenario, the PGPNP service offers a 'matchmaking' feature. It takes into account mutual connections between users when evaluating query consumption. If two Celo users both have each other in their respective address book, then they are informed of the connection without their quota being reduced. 
+
+## Decentralized PGPNP
+### Distributed Key Generation
+For the sake of user privacy, no single party should have the ability to unilaterally compute the hash for a given number. With this in mind, we've designed the PGPNP service to be decentralized across a handful of reputable participants. Before the PGPNP service is deployed, a set of operators participated in a Distributed Key Generation (DKG) ceremony to generate a shared secret. You can find the repository for the [DKG setup here](https://github.com/celo-org/celo-threshold-bls-rs). Each instance of the PGPNP service holds a share of the key which can be used to sign the response to the user. When enough of these signatures are combined, their combination can be used to derive the phone number salt. The number of key holders and threshold of signatures required are both configurable at the time of the DKG ceremony.
+
+### Blinding
+When the user queries a phone number, the mobile wallet first blinds the phone number locally. After the application receives the response, it unblinds it to retrieve the salt. This blinding process preserves the privacy of the mobile number such that the PGPNP cannot tell which number it's providing a salt for; reducing risk of targeted censorship and further increasing privacy.  
+
+### Combiner
+Due to the multi-service communication that needs to happen as part of the K of M signing, we've created a combiner service which performs the orchestration on the user's behalf. This is entirely optional to facilitate the orchestration and a given client can choose to operate their own combiner if desired. Because the combiner only receives the blinded phone number, the combiner cannot tell which number it's providing orchestrating services for. The combiner additionally validates the response from each signer to ensure a corrupt signer cannot corrupt the resulting salt.
```

Comment:
What do you mean by it being "entirely optional"?  As in, requesters could perform the combining on their own if they like?

Reply:
Yes, that's what I was trying to convey. Perhaps I can reword it to make it more clear: "Using the cLabs operated combiner is optional and a client can choose to operate/designate a separate combiner if desired."